test/OpenTelemetry.Tests/Logs/OpenTelemetryLoggerProviderTests.cs

```diff
@@ -84,4 +86,49 @@ public void VerifyOptionsCannotBeChangedAfterInit(bool initialValue)
         Assert.Equal(initialValue, provider.Options.IncludeScopes);
         Assert.Equal(initialValue, provider.Options.ParseStateValues);
     }
+
+    [Fact]
+    public void VerifyAddProcessorOverloadWithImplementationFactory()
+    {
+        // arrange
+        var options = new OpenTelemetryLoggerOptions();
+        var myProcessor = new MyProcessor();
+        var serviceProviderMock = new Mock<IServiceProvider>();
+
+        serviceProviderMock.Setup(x => x.GetService(typeof(BaseProcessor<LogRecord>))).Returns(myProcessor);
+        static BaseProcessor<LogRecord> ImplementationFactory(IServiceProvider x) => x.GetService<BaseProcessor<LogRecord>>();
+
+        // act
+        options.AddProcessor(ImplementationFactory);
+
+        // assert
+        Assert.Single(options.ProcessorFactories);
+        var processorFactory = options.ProcessorFactories[0];
+        var processor = processorFactory(serviceProviderMock.Object);
+        Assert.Equal(myProcessor, processor);
+    }
+
+    [Fact]
+    public void VerifyExceptionIsThrownWhenImplementationFactoryIsNull()
+    {
+        // arrange
+        Func<IServiceProvider, BaseProcessor<LogRecord>> implementationFactory = null;
+        var services = new ServiceCollection();
+        services.AddLogging(logging =>
+            logging.AddOpenTelemetry(
+                o =>
+                o.AddProcessor(implementationFactory)));
+
+        services.BuildServiceProvider();
```

Comment:
Remove this line seems like a dupe of the next one.

Reply:
removed!